lib/fxpay/products.js

```diff
@@ -8,38 +8,66 @@
   var utils = fxpay.getattr('utils');
   var API = fxpay.getattr('api').API;
 
-  exports.all = function products_all(callback) {
-    var promise = new Promise(function(resolve, reject) {
-      var allProducts = [];
 
-      var api = new API(settings.apiUrlBase);
-      var origin = utils.getSelfOrigin();
-      if (!origin) {
-        return reject(errors.InvalidApp('an origin is needed to get products'));
-      }
-      origin = encodeURIComponent(origin);
-      var url;
-
-      if (settings.fakeProducts) {
-        settings.log.warn('about to fetch fake products');
-        url = '/payments/stub-in-app-products/';
-      } else {
-        settings.log.info('about to fetch real products for app',
-                          origin);
-        url = '/payments/' + origin + '/in-app/?active=1';
-      }
+  exports.get = function products_get(productId, opt) {
+    return buildProductReceiptMap().then(function() {
+      return new Promise(function(resolve, reject) {
+        exports.getById(productId, function(error, product) {
+          if (error) {
+            settings.log.error(
+              'no existing product with productId=' + productId +
+              '; error: ' + error);
+            return reject(error);
+          }
+
+          resolve(product);
+        }, opt);
+      });
+    });
+  };
+
 
-      api.get(url, function(err, result) {
-        if (err) {
-          return reject(err);
+  exports.all = function products_all(callback) {
```

Comment:
I know this is refactored code, but in any case I'm not sure what the state of needing named funcs for debugging is so much any more - I know there was something on a webdev thread saying it was less necessary these days. Trying the first of the more complex examples from this article [1] the stack on FF + Chrome looked to work correctly - see [2]  and look at the console.

[1] http://kangax.github.io/nfe/#names-in-debuggers
[2] http://yakshavings.co.uk/js-debug/